docs/functions.md

```diff
@@ -0,0 +1,567 @@
+to be captured as a variable for reuse. The `customSpew` function is implemented
+by `spew.NewDefaultConfig`.
+
+> `NewDefaultConfig` returns a `spew.ConfigState` with the following default settings.
+
+> ```go
+>Indent: " "
+>MaxDepth: 0
+>DisableMethods: false
+>DisablePointerMethods: false
+>ContinueOnMethod: false
+>SortKeys: false
+>```
+
+##### Parameters
+
+- None
+
+##### Returns
+
+- `spew.ConfigState`, which is a customized Sprig printer suitable to be passed as
+    an argument to the customizing functions for further settings changes.
+
+##### Example
+
+Change the default indentation from one space to a tab and dump the current
+template context in place.
+
+```
+[[ $cs := ( customSpew | withIndent "  " ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withIndent` <a id="withIndent"></a>
+
+Sets the `Indent` flag for a customized Spew printer. From the Spew documentation:
+
+> `Indent` specifies the string to use for each indentation level.  The
+> global config instance that all top-level functions use sets this to a
+> single space by default.  If you would like more indentation, you might
+> set this to a tab with `"\t"` or perhaps two spaces with `"  "`.
+
+##### Parameters
+
+- 1: `string` - The value to set the `Indent` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+Change the default indentation from one space to a tab and dump the current
+template context in place.
+
+```
+[[ $cs := ( customSpew | withIndent "  " ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withMaxDepth` <a id="withMaxDepth"></a>
+
+Sets the `MaxDepth` flag for a customized Spew printer. From the Spew documentation:
+
+> `MaxDepth` controls the maximum number of levels to descend into nested
+> data structures.  The default, `0`, means there is no limit.
+>
+> **NOTE:** Circular data structures are properly detected, so it is not
+> necessary to set this value unless you specifically want to limit deeply
+> nested data structures.
+
+##### Parameters
+
+- 1: `int` - The value to set the `MaxDepth` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withMaxDepth true ) ]][[ $cs.Sdump . ]]
+```
+
+#### `withDisableMethods` <a id="withDisableMethods"></a>
+
+Sets the `DisableMethods` flag for a customized Spew printer. From the Spew documentation:
+
+> `DisableMethods` specifies whether or not `error` and `Stringer` interfaces are
+> invoked for types that implement them.
+
+##### Parameters
+
+- 1: `bool` - The value to set the `DisableMethods` value of the `s` parameter to.
+- 2: `spew.ConfigState` - A customized Sprig printer created by `customSpew`
+
+##### Returns
+
+- The modified Sprig printer
+
+##### Example
+
+```
+[[ $cs := ( customSpew | withDisableMethods true ) ]][[ $cs.Sdump . ]]
```

Comment:
withDisableMethods sets s.DisableMethods to true. So only "true" is not allowed (want 1 got 2). So it should be:
```
    [[ $cs := ( customSpew | withDisableMethods ) ]][[ $cs.Sdump . ]]
```

Reply:
Same with all the following, which use "true" as parameter